### PR TITLE
Add production shadow-grading runner

### PR DESCRIPTION
**Pull request by srinivasrowdur**
## Summary
- add an offline production shadow-grading runner that grades stored research traces without re-running live research
- reuse the product-resolution and guardrail-quality graders with trace-backed inputs and write JSON/Markdown outputs under `evals/reports/`
- add a local sample-trace fixture, runner tests, and README/package wiring for local development

## Validation
- npm run test:evals
- npm run build
- npm run evals:run-production-shadow-grading -- --input-file evals/cases/_fixtures/shadow-traces.sample.jsonl

Closes #23

### CODE REVIEW

File: evals/runProductionShadowGrading.ts [hunk 1]
```diff
@@ -0,0 +1,476 @@
...
+}
+
+function buildPreviousRecommendationMap(traces: StoredResearchRunTrace[]) {
+  const previousRecommendationByTrace = new Map<string, string | null>();
+  const previousBySubject = new Map<string, string | null>();
+
+  for (const trace of [...traces].sort((left, right) => compareCreatedAtAsc(left.createdAt, right.createdAt))) {
+    const subjectIdentity = trace.subjectKey ?? trace.requestedSubjectName.trim().toLowerCase();
+    previousRecommendationByTrace.set(trace.runId, previousBySubject.get(subjectIdentity) ?? null);
+    previousBySubject.set(subjectIdentity, trace.recommendation);
```
[the remark this answers is not in the document]
Comment (srinivasrowdur) [reply evals/runProductionShadowGrading.ts]: Agreed. I am fixing this by keeping the previous recommendation baseline append-only across failed or null-recommendation traces, so only succeeded gradeable runs with a non-null recommendation can advance the comparison baseline.

[the remark this answers is not in the document]
Comment (srinivasrowdur) [reply evals/runProductionShadowGrading.ts]: Fixed in `b958a62`.

`buildPreviousRecommendationMap(...)` now only advances the subject baseline on succeeded traces with a non-null recommendation, so failed/null-recommendation runs no longer clear the prior baseline. I also added a regression test for the exact `success(green) -> failed(null) -> success(red)` sequence.
